test/helper.cpp

```diff
@@ -5,18 +5,26 @@ using namespace matrix;
 
 int main()
 {
-    TEST(fabs(wrap_pi(4.0) - (4.0 - 2*M_PI)) < 1e-5);
-    TEST(fabs(wrap_pi(-4.0) - (-4.0 + 2*M_PI)) < 1e-5);
+    TEST(fabs(wrap(4.0, 0.0, 10.0) - 4.0) < 1e-5);
+    TEST(fabs(wrap(4.0, 0.0, 1.0)) < 1e-5);
+    TEST(fabs(wrap(-4.0, 0.0, 10.0) - 6.0) < 1e-5);
+    TEST(fabs(wrap(-18.0, 0.0, 10.0) - 2.0) < 1e-5);
+    TEST(fabs(wrap(-1.0, 30.0, 40.0) - 39.0) < 1e-5);
+    TEST(fabs(wrap(-8000.0, -555.0, 1.0) - (-216.0)) < 1e-5);
+
+    TEST(fabs(wrap_pi(4.0) - (4.0 - M_TWOPI)) < 1e-5);
+    TEST(fabs(wrap_pi(-4.0) - (-4.0 + M_TWOPI)) < 1e-5);
     TEST(fabs(wrap_pi(3.0) - (3.0)) < 1e-3);
-    TEST(!is_finite(wrap_pi(1000.0f)));
-    TEST(!is_finite(wrap_pi(-1000.0f)));
-    wrap_pi(NAN);
+    TEST(fabs(wrap_pi(1000.0f) - (1000.0f - 318 * M_PI)) < 1e-3);
+    TEST(fabs(wrap_pi(-1000.0f) - (-1000.0f + 318 * M_PI)) < 1e-3);
+    TEST(fabs(wrap_pi(-1001.0f) - (-1001.0f + 318 * M_PI)) < 1e-3);
+    TEST(!is_finite(wrap_pi(NAN)));
 
     TEST(fabs(wrap_2pi(-4.0) - (-4.0 + 2*M_PI)) < 1e-5);
     TEST(fabs(wrap_2pi(3.0) - (3.0)) < 1e-3);
-    TEST(!is_finite(wrap_2pi(1000.0f)));
-    TEST(!is_finite(wrap_2pi(-1000.0f)));
-    wrap_2pi(NAN);
+    TEST(fabs(wrap_2pi(2000.0f) - (2000.0f - 318 * M_TWOPI)) < 1e-3);
+    TEST(fabs(wrap_2pi(-2001.0f) - (-2001.0f + 319 * M_TWOPI)) < 1e-3);
```

Comment:
After performing the updated ops are all the data types still the same? Or would some vals be changing to doubles?